pkg/systemstatsmonitor/disk_collector.go

```diff
@@ -300,6 +316,34 @@ func listRootBlockDevices(timeout time.Duration) []string {
 	return strings.Split(strings.TrimSpace(string(stdout)), "\n")
 }
 
+// recordBootDiskSize records the disk block devices that are not a slave or holder.
+func (dc *diskCollector) recordBootDiskSize(timeout time.Duration) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	// "-d" prevents printing slave or holder devices. i.e. /dev/sda1, /dev/sda2...
+	// "-b" display the size in bytes.
+	// "-n" prevents printing the headings.
+	// "-p NAME" specifies to only print the device name.
+	cmd := exec.CommandContext(ctx, "lsblk", "-d", "-nb", "-o", "NAME,SIZE")
```

Comment:
I was looking into ways that don't fork to get this info:

```
$ time lsblk -d -nb -o NAME,SIZE 2>&1 >/dev/null

real    0m0.046s
user    0m0.025s
sys     0m0.027s

$ time cat /proc/partitions 2>&1 >/dev/null

real    0m0.045s
user    0m0.030s
sys     0m0.021s
```

Looks like they are comparable. Can you read from /proc instead of using lsblk? It may be a bit of work to get the boot disk only though.

Reply:
optimized the call to just one. hope that should be fine